app/src/main/java/org/mifos/mobile/repositories/UserAuthRepository.kt

```diff
@@ -15,4 +18,17 @@ interface UserAuthRepository {
         password: String?,
         username: String?
     ): Observable<ResponseBody?>?
+
+    fun login(username: String, password: String): Observable<User?>?
+
+    fun loadClient() : Observable<Page<Client?>?>?
+
+    fun saveAuthenticationTokenForSession(user: User)
+
+    fun reInitializeService()
+
+    fun setClientId(clientId: Long?)
```

Comment:
It is recommended to avoid having these functions within the UserAuthRepository class as it goes against the Single Responsibility Principle (SRP). To address this concern, it is great to create a new interface called `ClientRepository` along with its implementation, `ClientRepositoryImp`. This new class can then be used in the `LoginViewModel`.

Reply:
Got it, so insightful. made the req changes. 😃